src/ast/ast.cpp

```diff
@@ -388,5 +369,80 @@ void Probe::set_index(int index) {
   index_ = index;
 }
 
+Expression::Expression(const Expression &other) : Node(other)
+{
+  type = other.type;
+  is_literal = other.is_literal;
+  is_variable = other.is_variable;
+  is_map = other.is_map;
+}
+
+Call::Call(const Call &other) : Expression(other)
+{
+  func = other.func;
+}
+
+Binop::Binop(const Binop &other) : Expression(other)
+{
+  left = nullptr;
```

Comment:
just wonder why the only binop assign nullptr explicitly?

Reply:
left over from when the ast didnt have a default value yet, ill remove these